### ISSUE TITLE
High CPU load during download

### ISSUE THREAD

**Issue by genodeftest** [FedoraQt/MediaWriter#246]
During download of the fedora image, mediawriter has a very high CPU load of almost 100% on one core.

Steps to reproduce:
1. Open mediawriter
2. Download a fedora image
3. During download, watch CPU load

What happens:
High CPU load of >80% on one core

What should happen:
Almost no CPU load at all. `wget` can do the same with way less CPU load.

Version info:
mediawriter-4.1.4-4.fc32.x86_64

**Comment by grulja** [comment]
Is this still a problem with FMW 4.2.1?

**Comment by genodeftest** [comment]
Yes, this is still a problem (btw: Even when idle, the application eats quite a lot of CPU ressources, constantly running around 10% on my CPU)

Tested with mediawriter-4.2.1-2.fc33.x86_64 on Fedora 33 / GNOME+Wayland session

**Comment by grulja** [comment]
Closing now. Please re-open if this is still a problem with the latest Fedora Media Writer.

**Comment by genodeftest** [comment]
Problem is still present with `mediawriter-5.2.5-1.fc42.x86_64` (I cannot re-open this issue as I seem to lack permissions to do so)
